src/content/docs/dev/general-message-passing/sui/gmp-tutorial.mdx

```diff
@@ -0,0 +1,918 @@
+1. `params: vector<u8>`: For future use, this parameter is currently unused
+
+The function signature can be written out as follows: 
+
+```rust
+public fun send_message(
+    singleton: &Singleton,
+    gateway: &Gateway,
+    gas_service: &mut GasService,
+    destination_chain: String,
+    destination_address: String,
+    payload: vector<u8>,
+    refund_address: address,
+    coin: Coin<SUI>,
+    params: vector<u8>,
+) {}
+```
+
+Note: the `gas_service` param is of specified as mutable because you will eventually update the `GasService` resource’s internal state (by consuming the provided coin).
+
+You can now begin to implement the functionality for `send_message()`.
+
+First you can create a new [Message Ticket](/dev/general-message-passing/sui/gmp-example/#message-ticket). This can be interacting with the `Gateway's` `prepare_message()` function. 
+ 
+```rust
+let message_ticket = gateway::prepare_message(
+    &singleton.channel,
+    destination_chain,
+    destination_address,
+    payload,
+);
+```
+
+The ticket will contain all the relevant parameters needed when triggering a cross-chain message.
+
+Next, you can interact with the `GasService` to pay for the entirety of the cross chain call from the source chain. This can be done by triggering the `pay_gas()` function. The gas payment will be picked up by an Axelar relayer and eventually used to pay for the [execution fee](/dev/gas-service/pay-gas/#execution-fee) on the destination chain. 
+
+```rust
+gas_service.pay_gas(
+    &message_ticket,
+    coin,
+    refund_address,
+    params,
+);
+```
+
+Lastly, you can trigger the Gateway's `send_message()` function to trigger the cross-chain message by passing in the `message_ticket` you created earlier. 
+
+
+```rust
+gateway.send_message(message_ticket);
+```
+
+The complete `send_message()` function should look like this:
+
+```rust
+public fun send_message(
+    singleton: &Singleton,
+    gateway: &Gateway,
+    gas_service: &mut GasService,
+    destination_chain: String,
+    destination_address: String,
+    payload: vector<u8>,
+    refund_address: address,
+    coin: Coin<SUI>,
+    params: vector<u8>,
+) {
+    // Create a new message ticket
+    let message_ticket = gateway::prepare_message(
+        &singleton.channel,
+        destination_chain,
+        destination_address,
+        payload,
+    );
+
+    // Pay for the gas
+    gas_service.pay_gas(
+        &message_ticket,
+        coin,
+        refund_address,
+        params,
+    );
+
+    // Dispatch the cross-chain message
+    gateway.send_message(message_ticket);
+}
+```
+
+### Receive Message
+
+With the package now able to send cross-chain messages, it would be great to also receive messages from other blockchains. This can be done by creating an `execute()` function that will be triggered by our Axelar's relaying infrastructure for any incoming message. The full function signature for the `execute()` function will require two parameters:
```

Comment:
```suggestion
With the package now able to send cross-chain messages, it would be great also to receive messages from other blockchains. This can be done by creating an `execute()` function triggered by Axelar's relaying infrastructure for any incoming message. The full function signature for the `execute()` function will require two parameters:
```